targets/TARGET_STM/qspi_api.c

```diff
@@ -490,8 +485,46 @@ qspi_status_t qspi_init(qspi_t *obj, PinName io0, PinName io1, PinName io2, PinN
 
     return qspi_frequency(obj, hz);
 }
-#else /* OCTOSPI */
+
 qspi_status_t qspi_init(qspi_t *obj, PinName io0, PinName io1, PinName io2, PinName io3, PinName sclk, PinName ssel, uint32_t hz, uint8_t mode)
+{
+    QSPIName qspiio0name = (QSPIName)pinmap_peripheral(io0, PinMap_QSPI_DATA0);
+    QSPIName qspiio1name = (QSPIName)pinmap_peripheral(io1, PinMap_QSPI_DATA1);
+    QSPIName qspiio2name = (QSPIName)pinmap_peripheral(io2, PinMap_QSPI_DATA2);
+    QSPIName qspiio3name = (QSPIName)pinmap_peripheral(io3, PinMap_QSPI_DATA3);
+    QSPIName qspiclkname = (QSPIName)pinmap_peripheral(sclk, PinMap_QSPI_SCLK);
+    QSPIName qspisselname = (QSPIName)pinmap_peripheral(ssel, PinMap_QSPI_SSEL);
+
+    QSPIName qspi_data_first = (QSPIName)pinmap_merge(qspiio0name, qspiio1name);
+    QSPIName qspi_data_second = (QSPIName)pinmap_merge(qspiio2name, qspiio3name);
+    QSPIName qspi_data_third = (QSPIName)pinmap_merge(qspiclkname, qspisselname);
+
+    if (qspi_data_first != qspi_data_second || qspi_data_second != qspi_data_third ||
+            qspi_data_first != qspi_data_third) {
+        return QSPI_STATUS_INVALID_PARAMETER;
+    }
+
+    int peripheral = (int)qspi_data_first;
+    int function_io0 = (int)pinmap_find_function(io0, PinMap_QSPI_DATA0);
+    int function_io1 = (int)pinmap_find_function(io1, PinMap_QSPI_DATA1);
+    int function_io2 = (int)pinmap_find_function(io2, PinMap_QSPI_DATA2);
+    int function_io3 = (int)pinmap_find_function(io3, PinMap_QSPI_DATA3);
+    int function_sclk = (int)pinmap_find_function(sclk, PinMap_QSPI_SCLK);
+    int function_ssel = (int)pinmap_find_function(ssel, PinMap_QSPI_SSEL);
+
+    const qspi_pinmap_t static_pinmap = {peripheral, io0, function_io0, io1, function_io1, io2, function_io2, io3, function_io3, sclk, function_sclk, ssel, function_ssel};
+
+    QSPI_INIT_DIRECT(obj, &static_pinmap, hz, mode);
```

Comment:
I can reproduce the failure of qspi test, I think that is the cause of storage test. `qspi_init` function requires a return value.
```suggestion
    return QSPI_INIT_DIRECT(obj, &static_pinmap, hz, mode);
```

Reply:
Nice catch! Thank you.

Reply:
Can we commit the suggestion ?